float64 move_x
float64 move_y